### PR TITLE
Update Discount Sample to CLI 3.0 and remove Makefiles

### PR DESCRIPTION
**Pull request by nickwesselman**
* Completes the update of the discount sample to CLI3 
* Removes Make and `wasm-opt` dependencies from both function scaffolding and the discount sample

### CODE REVIEW

File: discounts/rust/shipping-discounts/default/shopify.function.extension.toml.liquid [hunk 1]
```diff
@@ -1,5 +1,6 @@
 name = "{{name}}"
 type = "{{extensionType}}"
+buildWasmPath = "target/wasm32-wasi/release/{{name | replace: " ", "-" | downcase}}.wasm"
```
Comment (jacobsteves) [inline discounts/rust/shipping-discounts/default/shopify.function.extension.toml.liquid:3]: CLI 3.0 can read both camelCase and snake_case in the toml file, but I'd recommend sticking with snake_case for consistency

**Review by jacobsteves — APPROVED** [review]
I tried to tophat this but I don't know how to specify to use a branch in `function-examples`. But 🤷‍♂️ LGTM

**Comment by dphm** [comment]
@jacobsteves 

```shell
yarn create @shopify/app --template https://github.com/Shopify/scripts-apis-examples/sample-apps/discount-functions-sample-app#nw/remove-makefiles
```

File: discounts/rust/shipping-discounts/default/shopify.function.extension.toml.liquid (hunk 1 above)
Comment (nickwesselman) [reply discounts/rust/shipping-discounts/default/shopify.function.extension.toml.liquid:3]: For `buildWasmPath`? That's a built-in arg to the CLI now

File: sample-apps/discount-functions-sample-app/README.md [hunk 2]
```diff
@@ -12,14 +12,17 @@ It also provides [Shopify Functions](#) that allow merchants to set up discounts
...
+  - `export WASM_OPT=/opt/homebrew/bin/wasm-opt`
 
 ## Installation
 
-1. Using the [Shopify CLI](#), connect this code with an app on your partner account.
-2. For each function in `extensions`, push it to your app: `shopify script push`
-3. Rename the `.env.sample` file in `web/frontend` to `.env` and fill in the function UUIDs for each of your functions
-4. Run `yarn dev` to start the server
+1. `yarn deploy` from the app root. This will allow you to create a new app, if needed.
+2. Rename the `.env.sample` file in `web/frontend` to `.env` and fill in the function UUIDs for each of your functions from the root `.env`
```
[the remark this answers is not in the document]
Comment (nickwesselman) [reply sample-apps/discount-functions-sample-app/README.md:24]: Yeah agreed, save for a later enhancement?

File: sample-apps/discount-functions-sample-app/extensions/order-discount/shopify.function.extension.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name = "order-discount"
```
[the remark this answers is not in the document]
Comment (nickwesselman) [reply sample-apps/discount-functions-sample-app/extensions/order-discount/shopify.function.extension.toml:1]: Missed that one, thanks!

**Pull request merged**